cmd/otel-allocator/allocation/http.go

```diff
@@ -0,0 +1,109 @@
+package allocation
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/prometheus/common/model"
+)
+
+type TargetItem struct {
+	JobName   string
+	Link      linkJSON
+	TargetURL string
+	Label     model.LabelSet
+	Collector *collector
+}
+
+type linkJSON struct {
+	Link string `json:"_link"`
+}
+
+type collectorJSON struct {
+	Link string            `json:"_link"`
+	Jobs []targetGroupJSON `json:"targets"`
+}
+
+type targetGroupJSON struct {
+	Targets []string       `json:"targets"`
+	Labels  model.LabelSet `json:"labels"`
+}
+
+func (allocator *Allocator) JobHandler(w http.ResponseWriter, r *http.Request) {
+	displayData := make(map[string]linkJSON)
+	for _, v := range allocator.targetItems {
+		displayData[v.JobName] = linkJSON{v.Link.Link}
+	}
+	allocator.jsonHandler(w, r, displayData)
+}
+
+func (allocator *Allocator) TargetsHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
Suggestion: A large portion of the code in here is finding things within lists of things that match an id (e.g. find the targets with job ID foo.  Readability could probably be improved by helper functions, such as `getCollectorTarget(collector, job)`, or `getAllTargets(job)`